src/hooks/useEscHandler/index.ts

```diff
@@ -13,14 +13,21 @@ export const useEscHandler = (onEscKeyDown: () => void, parentElementId: string)
         }
 
         const parentElement = document.getElementById(parentElementId);
-
-        if (parentElement) {
-          if (parentElement.contains(activeElement)) {
-            // If the active element is a child of the specified parent element, do nothing
-            return;
-          } else if (activeElement.tagName === 'BODY') {
+        if (!parentElement) {
+          return;
+        }
+        if (parentElement.contains(activeElement)) {
+          const parentElementId = parentElement.id;
+          if (parentElementId && parentElementId.startsWith(':') && parentElementId.endsWith(':')) {
```

Comment:
Hmm.. I'm not sure why the `parentElementId` is made a new const as the same `id` should already be available as it is one of the required parameters of the method (line 20).
And therefore there should be no need for the check if the `parentElementId` exists (line 21)